code/modules/admin/verbs/freeze.dm

```diff
@@ -0,0 +1,97 @@
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+//////Freeze Mob/Mech Verb -- Ported from NSS Pheonix (Unbound Travels)/////////
+////////////////////////////////////////////////////////////////////////////////
+//////Allows admin's to right click on any mob/mech and freeze them in place.///
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+/client/proc/freeze(mob/living/M as mob in mob_list)
+	set category = "Special Verbs"
+	set name = "Freeze"
+	if(!holder)
+		src << "Only administrators may use this command."
+		return
+	if(!mob)
+		return
+	if(!istype(M))
+		alert("Cannot freeze a ghost")
+		return
+	if(usr)
+		if (usr.client)
+			if(usr.client.holder)
+				if (istype(M, /mob/living/carbon/slime))
+					if(!M.paralysis)
+						M.adjustToxLoss(2147483647)
+						M.AdjustParalysis(2147483647)
+						var/adminomaly = new/obj/effect/overlay/adminoverlay
+						spawn(50)
+							M.overlays += adminomaly
+						M << "<b><font color= red>You have been frozen by <a href='?priv_msg=\ref[usr.client]'>[key]</a></b></font>"
+						message_admins("\blue [key_name_admin(usr)] froze [key_name(M)]")
+						log_admin("[key_name(usr)] froze [key_name(M)]")
+					else if (M.paralysis)
+						M.AdjustParalysis(-2147483647)
+						M.blinded = 0
+						M.lying = 0
+						M.stat = 0
+						M << "<b> <font color= red>You have been unfrozen by <a href='?priv_msg=\ref[usr.client]'>[key]</a></b></font>"
+						message_admins("\blue [key_name_admin(usr)] unfroze [key_name(M)]")
+						log_admin("[key_name(usr)] unfroze [key_name(M)]")
+						M.revive()
+				else
+					if(!M.paralysis)
+						M.AdjustParalysis(2147483647)
+						var/adminomaly = new/obj/effect/overlay/adminoverlay
+						spawn(50)
+							M.overlays += adminomaly
+						M << "<b><font color= red>You have been frozen by <a href='?priv_msg=\ref[usr.client]'>[key]</a></b></font>"
+						message_admins("\blue [key_name_admin(usr)] froze [key_name(M)]")
+						log_admin("[key_name(usr)] froze [key_name(M)]")
+					else if (M.paralysis)
+						M.AdjustParalysis(-2147483647)
+						M.blinded = 0
+						M.lying = 0
+						M.stat = 0
+
+						M << "<b> <font color= red>You have been unfrozen by <a href='?priv_msg=\ref[usr.client]'>[key]</a></b></font>"
+						message_admins("\blue [key_name_admin(usr)] unfroze [key_name(M)]")
+						log_admin("[key_name(usr)] unfroze [key_name(M)]")
+
+
+/client/proc/freezemecha(obj/mecha/O as obj in world)
```

Comment:
I'd recommend replacing world with mecha_list. It should be much more efficient.